File: LeetCode/algorithms/easy/python/1.TwoSum.py
# python
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        table = dict()
        for index, value in enumerate(nums):
            if (table.get(target - value) != None):
                return [ table.get(target - value), index ]
            table[value] = index
        return []

# python3
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        table = dict()
        for index, value in enumerate(nums):
            if (table.get(target - value) != None):
                return [ table.get(target - value), index ]
            table[value] = index
        return []
    

File: LeetCode/algorithms/easy/java/1.TwoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> table = new HashMap<>();
        for (int i = 0; i < nums.length; ++i) {
            if (table.get(target - nums[i]) != null)
                return new int[] { table.get(target - nums[i]), i };
            table.put(nums[i], i);
        }
        return new int[] {};
    }
}
